server/player/player.go

```diff
@@ -531,7 +531,17 @@ func (p *Player) Hurt(dmg float64, source damage.Source) (float64, bool) {
 		return 0, false
 	}
 
-	fireSource := source == damage.SourceFire{} || source == damage.SourceFireTick{} || source == damage.SourceLava{}
+	fireSource := false
+	if _, ok := source.(damage.SourceFireTick); ok {
+		fireSource = true
+	}
+	if _, ok := source.(damage.SourceLava); ok {
+		fireSource = true
+	}
+	if _, ok := source.(damage.SourceFire); ok {
+		fireSource = true
+	}
```

Comment:
Would either add a method to the damage source interface to return if it's fire damage, or put this in a switch.